gensim/models/wrappers/fasttext.py

```diff
@@ -337,8 +367,26 @@ def init_ngrams(self):
         """
         self.wv.ngrams = {}
         all_ngrams = []
+        if bin_only:
+            self.wv.syn0 = zeros((len(self.wv.vocab), self.vector_size), dtype=REAL)
         for w, v in self.wv.vocab.items():
-            all_ngrams += self.compute_ngrams(w, self.wv.min_n, self.wv.max_n)
+            word_ngrams = self.compute_ngrams(w, self.wv.min_n, self.wv.max_n)
+            all_ngrams += word_ngrams
+
+            
+            if bin_only:
+                #self.wv.syn0 = zeros((len(self.wv.vocab), self.vector_size), dtype=REAL)
+                word_vec = np.zeros(self.wv.syn0.shape[1])
+
+                num_word_ngram_vectors = len(word_ngrams)
+                for word_ngram in word_ngrams:
```

Comment:
Okay, so from looking at the FastText code, it seems this is more complicated than what we originally thought.
The final in-vocab word vectors (aka the syn0 vectors, the vectors in the `.vec` file) are obtained by summing up - 
1. all vectors for its component ngrams
2. the vector for the original word itself 

The large weight matrix in the `.bin` file contains - 
1. vectors for the original words in the first `len(vocab)` rows
2. in the remaining rows, weights for component ngrams

This will require a change in logic then.

Possibly useful reference - https://github.com/facebookresearch/fastText/blob/master/src/dictionary.cc#L71